utils/utf8.cc

```diff
@@ -584,6 +586,30 @@ std::optional<size_t> validate_with_error_position(const uint8_t *data, size_t l
     return validate_naive(data, len);
 }
 
+// From https://stackoverflow.com/a/33799784/12786387
+sstring escape_json(const sstring &s) {
+    std::ostringstream o;
+    for (auto c = s.cbegin(); c != s.cend(); c++) {
+        switch (*c) {
+        case '"': o << "\\\""; break;
+        case '\\': o << "\\\\"; break;
+        case '\b': o << "\\b"; break;
+        case '\f': o << "\\f"; break;
+        case '\n': o << "\\n"; break;
+        case '\r': o << "\\r"; break;
+        case '\t': o << "\\t"; break;
+        default:
+            if ('\x00' <= *c && *c <= '\x1f') {
+                o << "\\u"
+                  << std::hex << std::setw(4) << std::setfill('0') << (int)*c;
```

Comment:
Note: both std::hex and std::setfill are sticky in the sense that they will effect any integer printout you do later on the same stream. So you can just do them once instead of once for each control character.
Setw is NOT sticky and needs to be done every time!